app/javascript/react/store/fixedSessionsSlice.ts

```diff
@@ -48,22 +51,54 @@ export interface SessionsData {
 
 const initialState: SessionsState = {
   fetchableSessionsCount: 0,
-  sessions: [],
+  activeSessions: [],
+  dormantSessions: [],
+  isActiveSessionsFetched: false,
+  isDormantSessionsFetched: false,
   status: StatusEnum.Idle,
   error: undefined,
 };
 
-export const fetchFixedSessions = createAsyncThunk<
+export const fetchActiveFixedSessions = createAsyncThunk<
   SessionsResponse,
   SessionsData,
   { rejectValue: string }
 >(
-  "sessions/fetchFixedSessions",
+  "sessions/fetchActiveFixedSessions",
   async (sessionsData, { rejectWithValue }) => {
     try {
       const response: AxiosResponse<SessionsResponse, Error> =
         await oldApiClient.get(
-          API_ENDPOINTS.fetchFixedSessions(sessionsData.filters)
+          API_ENDPOINTS.fetchActiveFixedSessions(sessionsData.filters)
+        );
+
+      return response.data;
+    } catch (error) {
+      const message = getErrorMessage(error);
+      return rejectWithValue(message);
+    }
+  },
+  {
+    condition: (_, { getState }) => {
+      const { fixedSessions } = getState() as RootState;
+      if (fixedSessions.status === StatusEnum.Pending) {
+        return false;
+      }
+    },
+  }
+);
+
+export const fetchDormantFixedSessions = createAsyncThunk<
+  SessionsResponse,
+  SessionsData,
+  { rejectValue: string }
+>(
+  "sessions/fetchDormantFixedSessions",
+  async (sessionsData, { rejectWithValue }) => {
+    try {
+      const response: AxiosResponse<SessionsResponse, Error> =
+        await oldApiClient.get(
+          API_ENDPOINTS.fetchDormantFixedSessions(sessionsData.filters)
         );
 
       return response.data;
```

Comment:
![theyre-the-same-picture-the-office](https://github.com/user-attachments/assets/31c97323-df12-4353-b445-83bcabb31163)

I'd just combine them both for the sake of maintainabilty. 

```
const createSessionFetchThunk = (type: 'active' | 'dormant', endpoint: (filters: string) => string) => {
  return createAsyncThunk<SessionsResponse, SessionsData, { rejectValue: string }>(
    `sessions/fetch${type.charAt(0).toUpperCase() + type.slice(1)}FixedSessions`,
    async (sessionsData, { rejectWithValue }) => {
      [...]
};
```

And then use them separately:
```
export const fetchActiveFixedSessions = createSessionFetchThunk('active', API_ENDPOINTS.fetchActiveFixedSessions);
export const fetchDormantFixedSessions = createSessionFetchThunk('dormant', API_ENDPOINTS.fetchDormantFixedSessions);
```

Reply:
Cooool! Thank you!!

Reply:
@bazelajoanna nice, we still need to change the api client to use the endpoints flexibly: `oldApiClient.get(endpoint(sessionsData.filters))`, because now you're only using active fiixed sessions:
`API_ENDPOINTS.fetchActiveFixedSessions(sessionsData.filters)`